Dependencies/Authentication/Sources/Authentication/AuthenticationManager.swift

```diff
@@ -0,0 +1,185 @@
+    private let auth = Auth.auth()
+    
+    
+    #warning("Not finished yet")
+//    private var userHandler: ((AuthenticationInterface.User?) -> Void)?
+//        lazy var userUpdates: AsyncStream<AuthenticationInterface.User?> = {
+//            AsyncStream(bufferingPolicy: .bufferingNewest(1)) { continuation in
+//                userHandler = { continuation.yield($0) }
+//            }
+//        }()
+    
+    private var user: AuthenticationInterface.User? {
+        didSet {
+            //userHandler?(user)
+            loginEventHandler?(user != nil)
+        }
+    }
+    
+    lazy var signInResult: AsyncStream<Bool> = {
+        AsyncStream(bufferingPolicy: .bufferingNewest(1)) { continuation in
+            loginEventHandler = { continuation.yield($0) }
+        }
+    }()
+    
+    func getAuthenticatedUser() throws -> Bool {
+        guard let auth = auth.currentUser else {
+            throw AuthErrorHandler.getAuthenticatedUserError
+        }
+        return true
+    }
+    
+    // MARK: Sign In with email
+    func createUser(email: String, password: String) async throws {
+        do {
+            try await auth.createUser(withEmail: email, password: password)
+        } catch {
+            throw AuthErrorHandler.signUpError
+        }
+    }
+    
+    func signInUser(email: String, password: String) async throws {
+        do {
+            let authDataResult = try await auth.signIn(withEmail: email, password: password).user
+            self.user = User(from: authDataResult)
+        } catch {
+            throw AuthErrorHandler.signInError
+        }
+    }
+    
+    func updatePassword(email: String, password: String, newPassword: String) async throws {
+        guard let user = auth.currentUser else {
+            throw AuthErrorHandler.updatePasswordError
+        }
+        try await user.updatePassword(to: password)
+    }
+    
+    func resetPassword(email: String) async throws {
+        do {
+            try await auth.sendPasswordReset(withEmail: email)
+        } catch {
+            throw AuthErrorHandler.resetPasswordError
+        }
+    }
+    
+    func deleteAccount() async throws {
+        guard let user = auth.currentUser else {
+            throw AuthErrorHandler.deleteUserError
+        }
+        try await user.delete()
+    }
+    
+    // MARK: PROVIDERS
+    func getProviders() throws -> [AuthenticationInterface.AuthProviderOption] {
+        guard let providerData = auth.currentUser?.providerData else {
+            throw AuthErrorHandler.getProvidersError
+        }
+        
+        var providers: [AuthProviderOption] = []
+        for provider in providerData {
+            if let option = AuthProviderOption(rawValue: provider.providerID) {
+                providers.append(option)
+                print(option)
+            } else {
+                assertionFailure("Provicer option not found: \(provider.providerID)")
+            }
+        }
+        return providers
+    }
```

Comment:
```suggestion
   func getProviders() throws -> [AuthenticationInterface.AuthProviderOption] {
        guard let user = auth.currentUser else {
            throw AuthErrorHandler.getAuthenticatedUserError
        }
        
        return user
            .providerData
            .compactMap { AuthenticationInterface.AuthProviderOption(rawValue: $0.providerID) }
    }
```

Powinno działać tak samo